manifests/config.pp

```diff
@@ -50,6 +50,14 @@
     path => "${foreman_proxy::config_dir}/settings.yml",
   }
 
+  $uuid_fact_signature = foreman_proxy::hmac_signature($foreman_proxy::oauth_consumer_secret, $foreman_proxy::instance_id)
+  file { 'smart_proxy_uuid.json':
+    path    => '/opt/puppetlabs/facter/facts.d/smart_proxy_uuid.json',
+    owner   => $foreman_proxy::user,
```

Comment:
Why is it owned by `foreman-proxy`? Does that user ever need to write (or read?) that file?

Reply:
It felt like the correct user to use, although the user does neither write nor read that file.

Reply:
For those cases I generally recommend `root:$group` with mode 0640. The question is, which group? In this case, perhaps `root` is the most logical. If `foreman-proxy` doesn't read it.

Reply:
Yeah, I'd even go as far as setting it 0440 or 0400 as not even root really writes it.
(CAP_DAC_OVERRIDE is present when you run puppet as root, so that should work, untested).

Reply:
I know we use 440 / 400 somewhere with Puppet.

Reply:
Went with `root:root` and `0400`

Reply:
I think that comment is only relevant for Windows.

Reply:
That's possible. IIRC the change didn't help fix the tests. In any way, puppet-foreman and puppet-foreman_proxy seem to use 0440 so let's stick with that.